### DISCUSSION TITLE
find the dropped edges when dropout_edge force_undirected=True

### DISCUSSION THREAD

**Discussion by cxw-droid** [pyg-team/pytorch_geometric#8371] [Q&A]
Hi, how can I find the dropped edges after using `dropout_edge`?  If `force_undirected=False`, it is easy:
`dropped_edges = original_edge_index[:, ~edge_mask]`
But if `force_undirected=True`, how can I find the dropped edges? Thanks.

**Comment by rusty1s** [comment]
I think you can still use `edge_mask` for this purpose, right? It is just that it only contains the upper triangle of edges that have been dropped.

**Reply by cxw-droid** [reply]
Thank you for your reply. If `force_undirected=True`, the returned `edge_mask` seems only masking the **un-dropped** nodes. So it seems not clear how to get the dropped nodes.

**Reply by rusty1s** [reply]
Ah, got it. This is indeed not really elegant (especially since `edge_mask` returns a `LongTensor`). Let me try to fix it.

**Reply by rusty1s** [reply]
Sorry for late reply. On a second look, this seems pretty difficult to implement. I suggest to get the dropped edges via
```python
dropped_edge_mask = torch.ones(edge_index.size(1), dtype=torch.bool)
dropped_edge_mask[edge_mask] = False
```

**Reply by cxw-droid** [reply]
Thank you for your reply. The second line of code may not be correct if `force_undirected=True`. `edge_mask` only shows the edges in one direction. The edges in the opposition directions are also kept (not dropped).

**Reply by EdisonLeeeee** [reply]
> edge_mask only shows the edges in one direction. 

Yes this is intended because the opposite edges may not exist in the input `edge_index`. So how about using `to_undirected` after `dropout_edge`?
```python
edge_index, mask = dropout_edge(edge_index)
edge_index, mask = to_undirected(edge_index, mask)
```
